src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -36,20 +47,72 @@ class SymbolInfoVisitor : public llvm::codeview::SymbolVisitorCallbacks {
     SymbolInfo symbol_info;
     symbol_info.set_name(proc.Name.str());
     symbol_info.set_demangled_name(llvm::demangle(proc.Name.str()));
+
+    // The ProcSym's name does not contain an argument list. However, this information is required
+    // when dealing with overloads and it is available in the type info stream. See:
+    // https://llvm.org/docs/PDB/TpiStream.html
+    llvm::StringRef argument_list = RetrieveArgumentList(proc);
+    if (!argument_list.empty()) {
+      symbol_info.set_demangled_name(
+          absl::StrCat(symbol_info.demangled_name(), argument_list.data()));
+    }
+
     // The address in PDB files is a relative virtual address (RVA), to make the address compatible
     // with how we do the computation, we need to add both the load bias (ImageBase for COFF) and
     // the offset of the executable section.
     symbol_info.set_address(proc.CodeOffset + object_file_info_.load_bias +
                             object_file_info_.executable_segment_offset);
     symbol_info.set_size(proc.CodeSize);
+
     *(module_symbols_->add_symbol_infos()) = std::move(symbol_info);
 
     return llvm::Error::success();
   }
 
  private:
+  [[nodiscard]] llvm::StringRef RetrieveArgumentList(const llvm::codeview::ProcSym& proc) const {
+    llvm::codeview::LazyRandomTypeCollection& type_collection = type_info_stream_->typeCollection();
+
+    // We expect function types being either LF_PROCEDURE or LF_MFUNCTION, which are non-simple
+    // types. However, there are cases where the function type is "<no type>. In those cases, we
+    // can't retrieve the argument list.
+    if (proc.FunctionType.isSimple()) {
+      llvm::StringRef function_type = type_collection.getTypeName(proc.FunctionType);
+      ORBIT_ERROR(
+          "Unable to retrieve parameter list for function \"%s\"; The function type is \"%s\"",
+          proc.Name, function_type);
+      return "";
+    }
+
+    llvm::codeview::CVType function_type = type_info_stream_->getType(proc.FunctionType);
+    switch (function_type.kind()) {
+      case llvm::codeview::LF_PROCEDURE: {
+        llvm::codeview::ProcedureRecord procedure_record;
+        llvm::Error error =
+            llvm::codeview::TypeDeserializer::deserializeAs<llvm::codeview::ProcedureRecord>(
+                function_type, procedure_record);
+        ORBIT_CHECK(!error);
+
+        llvm::StringRef parameter_list = type_collection.getTypeName(procedure_record.ArgumentList);
+        return parameter_list;
+      }
+      case llvm::codeview::LF_MFUNCTION: {
+        llvm::codeview::MemberFunctionRecord member_function_record;
+        llvm::Error error =
+            llvm::codeview::TypeDeserializer::deserializeAs<llvm::codeview::MemberFunctionRecord>(
+                function_type, member_function_record);
+        ORBIT_CHECK(!error);
+
+        return type_collection.getTypeName(member_function_record.ArgumentList);
+      }
+      default:
+        ORBIT_UNREACHABLE();
```

Comment:
Just as question.. we are sure the other cases cannot happen? Because of the isSimple check, or because proc.FunctionType is used, or both?

Reply:
I went through the cases here, and that should be it. I guess the important thing is, that we do want to get noticed if we miss a case here.

Maybe briefly about the background here. We are retrieving the debug information from the `PrcoSym` elements, which are descriptions about functions. Those have a member `FunctionType`, that describes the type of the function, i.e. its return type, parameter types, const, etc... For each such function type there is a unique entry in the PDB file. Type information can be in general found in the type stream (https://llvm.org/docs/PDB/TpiStream.html) and the only types used to describe functions are LF_MFUNCTION for member functions and LF_PROCEDURE -- with the special case of <no-type> as above.